pages/dashboard.tsx

```diff
@@ -0,0 +1,238 @@
+          }}
+        >
+          Step Up WebAuthn
+        </Button>
+      </>
+    );
+  }
+
+  return (
+    <Box display="flex" flexDirection="column" alignItems="center" justifyContent="center" gap={2}>
+      <h3>You need to step up {type} before creating Passkeys!</h3>
+      <TextField label="Enter OTP" variant="outlined" value={inputValue} onChange={handleInputChange} />
+      <Button variant="contained" color="primary" onClick={handleSendOTPButtonClick}>
+        Send OTP to {type === StepUpType.email ? user?.emails?.at(0).email : user?.phone_numbers?.at(0).phone_number}
+      </Button>
+      <Button variant="contained" color="primary" onClick={validateOTPButtonClick}>
+        Validate OTP
+      </Button>
+      {err}
+    </Box>
+  );
+};
+
+const RegisterComponent = ({ numPasskeys }: { numPasskeys: number }) => {
+  const [displayRegisterPasskey, setDisplayRegisterPasskey] = useState(false);
+
+  return (
+    <>
+      {displayRegisterPasskey ? (
+        <StytchPasskeyRegistration
+          config={{
+            products: [Products.emailMagicLinks],
+            emailMagicLinksOptions: {
+              loginRedirectURL: 'http://localhost:3000/?type=eml',
+              signupRedirectURL: 'http://localhost:3000/?type=eml',
+            },
+          }}
+          callbacks={{
+            onEvent: ({ type, data }) => {
+              if (type === StytchEventType.PasskeyDone || type === StytchEventType.PasskeySkip) {
+                // eslint-disable-next-line no-console
+                console.log('Passkey dismissed', data);
+                setDisplayRegisterPasskey(false);
+              }
+            },
+          }}
+        />
+      ) : (
+        <>
+          <Typography variant="caption">You have {numPasskeys} registered Passkey(s)</Typography>
+          <Button onClick={() => setDisplayRegisterPasskey(true)}>Create a Passkey</Button>
+        </>
+      )}
+    </>
+  );
+};
+
+function Dashboard() {
+  const router = useRouter();
+  const stytch = useStytch();
+  const { user, isInitialized } = useStytchUser();
+
+  const { session } = useStytchSession();
+  const [sessionHasWebauthnFactor, setSessionHasWebauthnFactor] = useState(false);
+  const [sessionHasPhoneFactor, setSessionHasPhoneFactor] = useState(false);
+  const [sessionHasEmailFactor, setSessionHasEmailFactor] = useState(false);
+  const shouldPromptWebauthn =
+    ((sessionHasPhoneFactor && !sessionHasWebauthnFactor) || (sessionHasEmailFactor && !sessionHasWebauthnFactor)) &&
+    user?.webauthn_registrations?.length > 0;
+  const shouldPromptPhone = !sessionHasPhoneFactor && sessionHasWebauthnFactor && user?.phone_numbers?.length > 0;
+  const shouldPromptEmail = !sessionHasEmailFactor && sessionHasWebauthnFactor && user?.emails?.length > 0;
+
+  useEffect(() => {
+    if (isInitialized && user === null) {
+      router.push('/login');
+    } else if (session) {
+      session.authentication_factors.forEach((factor: AuthenticationFactor) => {
```

Comment:
these shouldn't be in the state - they can be calculated directly from useStytchSession

Reply:
e.g. 
```
const sessionHasPhoneFactor = !!session?.authentication_factors.find(factor => factor.delivery_method === 'sms')
```